app/src/main/java/org/jellyfin/mobile/car/MediaService.kt

```diff
@@ -0,0 +1,609 @@
+     * Libraries is the initial view
+     */
+    private enum class MediaItemType {
+        Libraries,
+        Library,
+        LibraryLatest,
+        LibraryAlbums,
+        LibraryArtists,
+        LibrarySongs,
+        LibraryGenres,
+        LibraryPlaylists,
+        Album,
+        Artist,
+        Shuffle,
+        Playlist,
+    }
+
+    override fun onCreate() {
+        super.onCreate()
+
+        mediaSession = MediaSessionCompat(this, "MediaService").apply {
+            isActive = true
+        }
+
+        sessionToken = mediaSession.sessionToken
+
+        mediaController = MediaControllerCompat(this, mediaSession)
+
+        mediaSessionConnector = MediaSessionConnector(mediaSession).also {
+            val dataSourceFactory: DataSource.Factory =
+                DefaultDataSourceFactory(this@MediaService, "Jellyfin Android")
+
+            it.setPlayer(exoPlayer)
+            it.setPlaybackPreparer(
+                MediaPlaybackPreparer(
+                    exoPlayer,
+                    dataSourceFactory,
+                    appPreferences
+                )
+            )
+            it.setQueueNavigator(MediaQueueNavigator(mediaSession))
+        }
+    }
+
+    override fun onDestroy() {
+        mediaSession.run {
+            isActive = false
+            release()
+        }
+
+        exoPlayer.release()
+    }
+
+    override fun onGetRoot(
+        clientPackageName: String,
+        clientUid: Int,
+        rootHints: Bundle?
+    ): BrowserRoot? {
+        val rootExtras = Bundle().apply {
+            putBoolean(CONTENT_STYLE_SUPPORTED, true)
+            putInt(CONTENT_STYLE_BROWSABLE_HINT, CONTENT_STYLE_LIST_ITEM_HINT_VALUE)
+            putInt(CONTENT_STYLE_PLAYABLE_HINT, CONTENT_STYLE_LIST_ITEM_HINT_VALUE)
+        }
+
+        return BrowserRoot(MediaItemType.Libraries.toString(), rootExtras)
+    }
+
+    /**
+     * The parent id will be in various formats such as
+     * Libraries
+     * Library|{id}
+     * LibraryAlbums|{id}
+     * Album|{id}
+     */
+    override fun onLoadChildren(parentId: String, result: Result<MutableList<MediaItem>>) {
+        if (appPreferences.instanceUrl == null || appPreferences.instanceAccessToken == null || appPreferences.instanceUserId == null) {
+            // the required properties for calling the api client are not available
+            // this should only occur if the user does not have a server connection in the app
+            result.sendError(null)
+            return
+        }
+
+        // ensure the api client is up to date
+        apiClient.ChangeServerLocation(appPreferences.instanceUrl)
+        apiClient.SetAuthenticationInfo(
+            appPreferences.instanceAccessToken,
+            appPreferences.instanceUserId
+        )
+
+        when (parentId) {
+            /**
+             * View that shows all the music libraries
+             */
+            MediaItemType.Libraries.toString() -> {
+                apiClient.GetUserViews(
```

Comment:
Please add an extension function in `utils/ApiExtensions.kt` to use coroutines instead of callbacks for the GetUserViews function.

Reply:
(Same for other used api functions)

Reply:
> Please add an extension function in `utils/ApiExtensions.kt` to use coroutines instead of callbacks for the GetUserViews function.

Wrt this: I think we shouldn't merge this before we added the new api client in, so maybe it makes sense to wait a bit when it has to be updated/changed anyways at a later point.

Reply:
I'm not sure when the new apiclient is ready so I'd rather merge this when it's ready and 2.0.0 is released instead of waiting for the apiclient.

Reply:
I added the api extensions. Since `onLoadChildren` is not a suspend function, I had to add a `CoroutineScope` to media service. Let me know if there's a better way to do that. It seems to be working well so it's probably okay.